src/components/Onboarding/OnboardingContext.tsx

```diff
@@ -0,0 +1,197 @@
+    route: baseRoute + "edit-profile",
+  },
+  2: {
+    step: "Set your availability",
+    route: baseRoute + "availability",
+  },
+});
+
+const MenteeOnboardingText = (baseRoute: string) => ({
+  1: {
+    step: "Fill out your profile",
+    route: baseRoute + "edit-profile",
+  },
+  2: {
+    step: "Browse through available mentors",
+    route: baseRoute + "mentors",
+  },
+});
+
+const authLevelToText = (authLevel: AuthorizationLevel) => {
+  switch (authLevel) {
+    case AuthorizationLevel.Admin:
+      return AdminOnboardingText;
+    case AuthorizationLevel.Mentor:
+      return MentorOnboardingText;
+    default:
+      return MenteeOnboardingText;
+  }
+};
+
+interface OnboardingContextType {
+  switchingRoutes: boolean;
+  OnboardingComponent: JSX.Element;
+}
+
+const OnboardingContext = createContext<OnboardingContextType | undefined>(
+  undefined
+);
+
+const useOnboardingProvider = () => {
+  const currentProfile = useCurrentProfile();
+  const [updateProfile] = useUpdateProfileMutation({
+    refetchQueries: ["getMyUser"],
+  });
+
+  const authorizationLevel = useAuthorizationLevel();
+  const [loading, setLoading] = useState(false);
+  const [currentStep, setCurrentStep] = useState<number>(1);
+  const router = useRouter();
+
+  const MAX_STEPS = authorizationLevelToMaxSteps(authorizationLevel);
+  const baseRoute = `/program/${router.query.slug}/${router.query.profileRoute}/`;
+
+  const onFinish = () => {
+    const updateProfileInput: UpdateProfileInput = {
+      ...currentProfile.currentProfile,
+      showOnboarding: false,
+    };
+    updateProfile({
+      variables: {
+        profileId: currentProfile.currentProfile!.profileId,
+        data: updateProfileInput,
+      },
+    })
+      .then(() => {
+        currentProfile.refetchCurrentProfile!();
+        LocalStorage.delete("Onboarding Step");
+      })
+      .catch((err) => console.error(err));
+  };
+
+  const props: OnboardingProps = {
+    currentStep,
+    setCurrentStep: (num: number) => {
+      LocalStorage.set("Onboarding Step", num);
+      setCurrentStep(num);
+    },
+    loading,
+    setLoading,
+    onboardingText: authLevelToText(authorizationLevel)(baseRoute),
+    MAX_STEPS,
+    onFinish,
+  };
+
+  const onboardingStep = LocalStorage.get("Onboarding Step");
+  useEffect(() => {
+    if (onboardingStep && typeof onboardingStep == "number") {
+      setCurrentStep(onboardingStep);
+    }
+  }, []);
+
+  if (
+    authorizationLevel !== AuthorizationLevel.Admin &&
+    router.asPath !== props.onboardingText[currentStep]["route"] &&
+    onboardingStep &&
+    typeof onboardingStep == "number"
+  ) {
+    router.push(props.onboardingText[onboardingStep]["route"]);
+    return {
+      switchingRoutes: true,
+      OnboardingComponent: <OnboardingComponent {...props} />,
+    };
+  }
+
+  return {
+    switchingRoutes: false,
+    OnboardingComponent: <OnboardingComponent {...props} />,
+  };
+};
+
+export const OnboardingProvider = ({
```

Comment:
Curious as to why this is its own provider, am noob to it

Reply:
I made it a provider because I wanted to share state among multiple pages, but I think it's also possible to make it a hook?

Reply:
Changed it to a hook :' )